src/ScheduleFunctions.cpp

```diff
@@ -1381,12 +1381,21 @@ class InjectFunctionRealization : public IRMutator {
                                                                   replacements, add_lets);
                 producer = inject_stmt(producer, produceDef, f.definition().schedule().fuse_level().level);
             }
+        }
+
+        bool some_updated = false;
+        for (size_t j = 0; j == 0 || some_updated; j++) {
+            some_updated = false;
+            for (auto iter = funcs.rbegin(); iter != funcs.rend(); iter++) {
+                const auto &f = *iter;
 
-            for (size_t j = 0; j < f.updates().size(); ++j) {
-                string defPrefix = f.name() + ".s" + std::to_string(j + 1) + ".";
-                const Definition &def = f.updates()[j];
-                const Stmt &updateDef = build_produce_definition(f, defPrefix, def, true, replacements, add_lets);
-                producer = inject_stmt(producer, updateDef, def.schedule().fuse_level().level);
+                if (j < f.updates().size()) {
```

Comment:
So does this interleave the update definitions of the compute_with things for all the stages that aren't actually compute_with each other? Worth a comment explaining the strategy.

Reply:
I think it will, yeah. There might be an alternative fix in RealizationOrder that adds an appropriate edge. But it wasn't obvious to me how to do that (whereas this was quick), and I'm pretty sure this interleaving preserves the actual values.

It would be good to really write down what compute_with is supposed to do / guarantee. And then write some tests for that.